packages/eslint-config/index.js

```diff
@@ -1,12 +1,24 @@
+const { hasConfig } = require('@spotify/web-scripts-utils');
+
+const hasReact = hasConfig([
+  { type: 'dependency', dependency: 'react' },
+  { type: 'dependency', dependency: 'react', dependencyType: 'peer' },
+]);
+const hasTypescript = hasConfig([
+  { type: 'dependency', dependency: 'typescript' },
+  { type: 'dependency', dependency: 'typescript', dependencyType: 'dev' },
+  { type: 'file', pattern: 'tsconfig.json' },
+]);
+
 module.exports = {
   extends: [
     '@spotify/eslint-config-base',
-    '@spotify/eslint-config-react',
-    '@spotify/eslint-config-typescript',
+    hasReact ? '@spotify/eslint-config-react' : null,
```

Comment:
based on the setup from `kcd-scripts`: https://github.com/kentcdodds/kcd-scripts/blob/b1ee9f87278e0cdaf5a444556d3e4d5ce1342bd3/src/config/eslintrc.js#L7